core/src/main/java/org/mule/runtime/core/internal/metadata/SimpleDataType.java

```diff
@@ -115,14 +117,12 @@ public boolean equals(Object o) {
       return false;
     }
 
-    // TODO MULE-9987 Fix this
-    // ANY_MIME_TYPE will match to a null or non-null value for MediaType
-    if ((this.mimeType == null && that.mimeType != null || that.mimeType == null && this.mimeType != null)
-        && !MediaType.ANY.matches(that.mimeType)) {
+    if ((this.mimeType == null && that.mimeType != null || that.mimeType == null && this.mimeType != null)) {
```

Comment:
use `Object.equals` to compare  the `mediaType`, use the getter instead of the attribute directly